tests/delete_regress/tests.py

```diff
@@ -255,9 +255,9 @@ def test_delete_proxy_pair(self):
         self.assertEqual(len(FooFileProxy.objects.all()), 0)
 
     def test_19187_values(self):
-        with self.assertRaises(TypeError):
+        with self.assertRaises(AssertionError):
```

Comment:
Fair enough, I'll turn the "Cannot call xyz after .values() or .values_list()" exceptions into `TypeError`, there's still some incompatibility for `only()` and `defer()` that raised a `NotImplementedError` I think it's fine.

Eventually we should turn all assertions into exceptions but that's worthy of a separate fix.